code/modules/shuttle/super_cruise/orbital_poi_generator/objective_types/vip_extraction.dm

```diff
@@ -59,14 +75,55 @@
 			created_human.equipOutfit(/datum/outfit/greytide)
 			antag_elligable = TRUE
+		if("soviet_admiral")
+			created_human.flavor_text = "Ivan Ivanov Ivanovich II, your superior, sent you to this Corporate Station in order to negotiate a deal between the Third Soviet Union and Nanotrasen,\
+			despite your inability of speaking galactic common, now weeks later, you wish you would've never come to this place, if you somehow manage to survive this, Ivan can be certain to feel your fist in his fat mug."
+			created_human.add_quirk(/datum/quirk/foreigner) //Blyat
+			antag_elligable = TRUE
 	if(antag_elligable)
 		if(prob(7))
 			created_human.mind.make_Traitor()
 		else if(prob(8))
 			created_human.mind.make_Changeling()
 	mob_to_recover = created_human
 	generated = TRUE
+//=====================
+//Dangerous VIP recovery
+//=====================
+/datum/orbital_objective/vip_recovery/generate_objective_stuff(turf/chosen_turf)
+	var/mob/living/carbon/human/created_human = new(chosen_turf)
+	//Maybe polling ghosts would be better than the shintience code
+	created_human.set_playable()
+	created_human.mind_initialize()
+	//Remove nearby dangers
+	for(var/mob/living/simple_animal/hostile/SA in range(10, created_human))
+		qdel(SA)
+	//Give them a space worthy suit
+	var/turf/open/T = locate() in shuffle(view(1, created_human))
+	if(T)
+		new /obj/item/clothing/suit/space/hardsuit/ancient(T)
+		new /obj/item/tank/internals/oxygen(T)
+		new /obj/item/clothing/mask/gas(T)
+		new /obj/item/storage/belt/utility/full(T)
+	switch(pickweight(list(
+		"marooned_syndicate_op" = 1,
+		"serial_killer" = 3,
+		"mobster" = 2)))
 
+		if("marooned_syndicate_op")
+			created_human.flavor_text = "It was supposed to be just another hit on just another Nanotrasen outpost, but in a moment of panic your squad left you behind and now \
+			you're stuck on this abandoned rust-heap, you're certain that Nanotrasen will dispatch a team to investigate what happened here, but you don't intend to let them find out."
+			created_human.equipOutfit(/datum/outfit/syndicate_op)
+			created_human.mind.add_antag_datum(/datum/antagonist/marooned_syndicate_op)
+		if("serial_killer")
+			created_human.flavor_text = "<span class='userdanger'>NO NOOO NOOOOOOO!!! IT CAN'T BE!! THEY ARE COMING FOR YOU, DON'T LET THEM GET YOU, KILL. THEM. ALL!!!"
+			created_human.equipOutfit(/datum/outfit/insane_killer)
+			created_human.mind.add_antag_datum(/datum/antagonist/insane_killer)
+		if("mobster")
+			created_human.flavor_text = "These damn Nanotrasen pigs wouldn't pay up for building their station in the territory of the mob, so they got what's coming to them,\
+			however, they sure did a number on your private space yacht, you barely managed to escape before the whole thing blew up, and now you're stranded here."
+			created_human.mind.add_antag_datum(/datum/antagonist/mobster)
+			created_human.equipOutfit(/datum/outfit/mobster)
```

Comment:
Could be an idea to move the created_human spawning to a seperate proc, then override it on the dangerous varient so that code isn't repeated.

Reply:
I am not sure what you mean with the code being repeated, the normal variant does it like this too.

Reply:
Line 49 to 107

Have a proc/create_vip_mob() that creates the mob. All safe VIPs can be antag elligable, anything not should be in the dangerous category.

Reply:
Also another idea what if we just made these ones an assassination objective so you don't have to revive an antag on station.

Reply:
> Also another idea what if we just made these ones an assassination objective so you don't have to revive an antag on station.

That would require there to always be a ghost to take up the role, not sure what we should do if nobody signed up to play the hunted person.

Reply:
Sorry Bacon, but I am simply not experienced enough to do what you're asking.

Reply:
Thats fine, didn't think about somebody not taking up the role.

Reply:
That was in regards to creating a new proc that handles the create of the mob, by the way.

Reply:
```
/datum/objective/.../proc/create_vip()
	switch(pickweight(list(
		"marooned_syndicate_op" = 1,
		"serial_killer" = 3,
		"mobster" = 2)))

		if("marooned_syndicate_op")
			created_human.flavor_text = "It was supposed to be just another hit on just another Nanotrasen outpost, but in a moment of panic your squad left you behind and now \
			you're stuck on this abandoned rust-heap, you're certain that Nanotrasen will dispatch a team to investigate what happened here, but you don't intend to let them find out."
			created_human.equipOutfit(/datum/outfit/syndicate_op)
			created_human.mind.add_antag_datum(/datum/antagonist/marooned_syndicate_op)
		...........
```
then call create_vip() in place of where the switch is. Then override the create_vip proc to create the new dangerous VIPs

Reply:
Ohhhh.

Reply:
I think I may be stupid, I just keep ending up with undefined procs.
`		new /obj/item/clothing/suit/space/hardsuit/ancient(T)
		new /obj/item/tank/internals/oxygen(T)
		new /obj/item/clothing/mask/gas(T)
		new /obj/item/storage/belt/utility/full(T)
/datum/orbital_objective/vip_recovery/generate_objective_stuff/proc/create_vip()
	var/antag_elligable = FALSE
	switch(pickweight(list(
		"centcom_official" = 4,
		"dictator" = 1,
		"greytide" = 3,
		"soviet_admiral" = 1)))

		if("centcom_official")
			created_human.flavor_text = "You are a Central Command Official on board of a badly damaged station. Making your way back to civilization to uncover the secrets you hold is \
				your top priority as far as Nanotrasen is concerned, but just surviving 1 more day is all you can ask for."
			created_human.equipOutfit(/datum/outfit/centcom_official_vip)
			antag_elligable = TRUE
		if("dictator")
			created_human.flavor_text = "It has been months since your regime fell. Once a hero, now just someone wishing that they will see the next sunrise. You know those \
				Nanotrasen pigs are after you, and will stop at nothing to capture you. All you want at this point is to get out and survive, however it is likely you will never leave \
				without being captured."
			created_human.equipOutfit(/datum/outfit/vip_dictator)
			created_human.mind.add_antag_datum(/datum/antagonist/vip_dictator)
			antag_elligable = TRUE
		if("greytide")
			created_human.flavor_text = "You are just a lonely assistant, on a lonely derelict station. You dream of going home, \
				but it would take another one of the miracles that kept you alive to get you home."
			created_human.equipOutfit(/datum/outfit/greytide)
			antag_elligable = TRUE
		if("soviet_ademiral")
			created_human.flavor_text = "Ivan Ivanov Ivanovich II, your superior, sent you to this Corporate Station in order to negotiate a deal between the Third Soviet Union and Nanotrasen,\
			despite your inability of speaking galactic common, now weeks later, you wish you would've never come to this place, if you somehow manage to survive this, Ivan can be certain to feel your fist in his fat mug."
			created_human.add_quirk(/datum/quirk/foreigner) //Blyat
			antag_elligable = TRUE
	if(antag_elligable)
		if(prob(7))
			created_human.mind.make_Traitor()
		else if(prob(8))
			created_human.mind.make_Changeling()
	mob_to_recover = created_human
	generated = TRUE
//=====================
//Dangerous VIP recovery
//=====================
/datum/orbital_objective/vip_recovery/dangerous/generate_objective_stuff(turf/chosen_turf)
	var/mob/living/carbon/human/created_human = new(chosen_turf)
	//Maybe polling ghosts would be better than the shintience code
	created_human.set_playable()
	created_human.mind_initialize()
	//Remove nearby dangers
	for(var/mob/living/simple_animal/hostile/SA in range(10, created_human))
		qdel(SA)
	//Give them a space worthy suit
	var/turf/open/T = locate() in shuffle(view(1, created_human))
	if(T)
		new /obj/item/clothing/suit/space/hardsuit/ancient(T)
		new /obj/item/tank/internals/oxygen(T)
		new /obj/item/clothing/mask/gas(T)
		new /obj/item/storage/belt/utility/full(T)
	create_vip()(pickweight(list(
		"marooned_syndicate_op" = 1,
		"serial_killer" = 3,
		"mobster" = 2)))`

Reply:
Add /proc before the proc definition on the main type, then on the subtype add /.../create_vip just without the proc. Then you can call create_vip(). It has to be defined on the supertype though.

Reply:
`/datum/orbital_objective/vip_recovery/generate_objective_stuff/proc/create_vip()
	create_vip(pickweight(list(
		"centcom_official" = 4,
		"dictator" = 1,
		"greytide" = 3,
		"soviet_admiral" = 1)))

		if("centcom_official")
			created_human.flavor_text = "You are a Central Command Official on board of a badly damaged station. Making your way back to civilization to uncover the secrets you hold is \
				your top priority as far as Nanotrasen is concerned, but just surviving 1 more day is all you can ask for."
			created_human.equipOutfit(/datum/outfit/centcom_official_vip)
			antag_elligable = TRUE
		if("dictator")
			created_human.flavor_text = "It has been months since your regime fell. Once a hero, now just someone wishing that they will see the next sunrise. You know those \
				Nanotrasen pigs are after you, and will stop at nothing to capture you. All you want at this point is to get out and survive, however it is likely you will never leave \
				without being captured."
			created_human.equipOutfit(/datum/outfit/vip_dictator)
			created_human.mind.add_antag_datum(/datum/antagonist/vip_dictator)
			antag_elligable = TRUE
		if("greytide")
			created_human.flavor_text = "You are just a lonely assistant, on a lonely derelict station. You dream of going home, \
				but it would take another one of the miracles that kept you alive to get you home."
			created_human.equipOutfit(/datum/outfit/greytide)
			antag_elligable = TRUE
		if("soviet_ademiral")
			created_human.flavor_text = "Ivan Ivanov Ivanovich II, your superior, sent you to this Corporate Station in order to negotiate a deal between the Third Soviet Union and Nanotrasen,\
			despite your inability of speaking galactic common, now weeks later, you wish you would've never come to this place, if you somehow manage to survive this, Ivan can be certain to feel your fist in his fat mug."
			created_human.add_quirk(/datum/quirk/foreigner) //Blyat
			antag_elligable = TRUE
	if(antag_elligable)
		if(prob(7))
			created_human.mind.make_Traitor()
		else if(prob(8))
			created_human.mind.make_Changeling()
	mob_to_recover = created_human
	generated = TRUE`
	
code\modules\shuttle\super_cruise\orbital_poi_generator\objective_types\vip_extraction.dm:64:error: proc definition not allowed inside another proc

Doing this gives me that error.

Reply:
Yeah, no idea how to do this, sorry.